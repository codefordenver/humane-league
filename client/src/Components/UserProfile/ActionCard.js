import React from 'react';

const ActionCard = ({ type, action }) => {
  return (
    <div className={`action-card ${type}`}>
      <h3>{action.title}</h3>
      <p>{action.description}</p>
    </div>
  );
}

export default ActionCard;