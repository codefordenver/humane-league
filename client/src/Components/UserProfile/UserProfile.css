.UserProfile {
  align-items: center;
  display: flex;
  flex-direction: column; }
  .UserProfile .user-achievements,
  .UserProfile .user-preferences {
    align-items: center;
    background-color: #F5F5F5;
    display: flex;
    flex-direction: column;
    margin: 20px 0;
    padding: 25px;
    width: 75%; }
    .UserProfile .user-achievements h2,
    .UserProfile .user-preferences h2 {
      margin: 20px; }
    .UserProfile .user-achievements .user-preferences-form,
    .UserProfile .user-preferences .user-preferences-form {
      display: flex;
      flex-direction: column;
      font-size: 20px;
      width: 200px; }
      .UserProfile .user-achievements .user-preferences-form label,
      .UserProfile .user-preferences .user-preferences-form label {
        align-items: center;
        display: flex;
        justify-content: space-between;
        margin: 5px 0; }
      .UserProfile .user-achievements .user-preferences-form button,
      .UserProfile .user-preferences .user-preferences-form button {
        background-color: #6CC24A;
        border: none;
        border-radius: 2px;
        color: white;
        cursor: pointer;
        margin: 20px 0;
        padding: 10px; }
