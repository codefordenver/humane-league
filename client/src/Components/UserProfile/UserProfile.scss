.UserProfile {
  align-items: center;
  display: flex;
  flex-direction: column;

  h1 {
    color: white;
    font-weight: normal;
    font-size: 36px;
    margin-bottom: 10px;
  }

  .action-count-description {
    color: white;
    font-size: 22px;
    margin-bottom: 20px;
  }

  .user-preferences,
  .user-profile-edit {
    align-items: center;
    background-color: white;
    box-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
    display: flex;
    flex-direction: column;
    margin: 20px 0;
    padding: 25px 50px;

    h2 {
      font-size: 32px;
      font-weight: normal;
      margin-bottom: 20px;
    }

    .user-preferences-form,
    .user-profile-form {
      display: flex;
      flex-direction: column;
      font-size: 20px;
      width: 200px;

      label {
        align-items: center;
        display: flex;
        justify-content: space-between;
        margin: 5px 0;
      }

      button {
        background-color: #6CC24A;
        border: none;
        border-radius: 2px;
        color: white;
        cursor: pointer;
        margin: 20px 0;
        padding: 10px;
      }
    }

    .user-profile-form {
      width: 400px;

      .text-input {
        padding: 5px;
        width: 325px;
      }
    }
  }
}