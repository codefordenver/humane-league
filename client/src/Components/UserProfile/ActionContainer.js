import React from 'react';
import ActionCard from './ActionCard';

const ActionContainer = () => {
  return (
    <div>
      I am an ActionContainer
      <ActionCard />
    </div>
  );
}

export default ActionContainer;
