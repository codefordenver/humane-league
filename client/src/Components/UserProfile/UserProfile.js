import React from 'react';
import ActionContainer from '../ActionContainer/ActionContainer';
import './UserProfile.css';

const UserProfile = () => {

  return (
    <div className="UserProfile">
      <h1 className="user-welcome">Welcome, Katie</h1>
      <ActionContainer/>
    </div>
  );
  
}

export default UserProfile;