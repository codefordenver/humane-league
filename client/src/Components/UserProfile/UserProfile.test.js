import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { UserProfile } from './UserProfile';

describe('UserProfile component tests', () => {
  let renderedUserProfile;
  let defaultState;
  let mockUser; 
  let userPreferences;
  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({ results: []})
    }));
    userPreferences =  { twitter_actions: false, facebook_actions: false, email_actions: false, phone_actions: false };
    mockUser = { id: 1, name: 'Julie Hawkins', email: 'Julabi@gmail.com', ...userPreferences };
    renderedUserProfile = shallow( <UserProfile user={mockUser}/> );
    defaultState = {
      name: 'Julie Hawkins',
      email: 'Julabi@gmail.com',
      actionCount: 0,
      twitter_actions: false,
      facebook_actions: false,
      email_actions: false,
      phone_actions: false
    };
  });
  it('renders without crashing', () => {
    expect(renderedUserProfile).toBeDefined();
    expect(renderedUserProfile.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedUserProfile).toMatchSnapshot();
  });
});