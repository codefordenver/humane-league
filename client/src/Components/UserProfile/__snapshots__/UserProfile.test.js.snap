// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component tests matches SnapShot 1`] = `undefined`;
